src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPClientFeatures.java

```diff
@@ -0,0 +1,609 @@
+        }
+        return semanticTokensFeature;
+    }
+
+    /**
+     * Initialize the LSP semanticTokens feature.
+     *
+     * @param semanticTokensFeature the LSP semanticTokens feature.
+     * @return the LSP client features.
+     */
+    public final LSPClientFeatures setSemanticTokensFeature(@NotNull LSPSemanticTokensFeature semanticTokensFeature) {
+        semanticTokensFeature.setClientFeatures(this);
+        this.semanticTokensFeature = semanticTokensFeature;
+        return this;
+    }
+
+    /**
+     * Returns the LSP signatureHelp feature.
+     *
+     * @return the LSP signatureHelp feature.
+     */
+    @NotNull
+    public final LSPSignatureHelpFeature getSignatureHelpFeature() {
+        if (signatureHelpFeature == null) {
+            setSignatureHelpFeature(new LSPSignatureHelpFeature());
+        }
+        return signatureHelpFeature;
+    }
+
+    /**
+     * Initialize the LSP signatureHelp feature.
+     *
+     * @param signatureHelpFeature the LSP signatureHelp feature.
+     * @return the LSP client features.
+     */
+    public final LSPClientFeatures setSignatureHelpFeature(@NotNull LSPSignatureHelpFeature signatureHelpFeature) {
+        signatureHelpFeature.setClientFeatures(this);
+        this.signatureHelpFeature = signatureHelpFeature;
+        return this;
+    }
+
+    /**
+     * Returns the LSP typeDefinition feature.
+     *
+     * @return the LSP typeDefinition feature.
+     */
+    @NotNull
+    public final LSPTypeDefinitionFeature getTypeDefinitionFeature() {
+        if (typeDefinitionFeature == null) {
+            setTypeDefinitionFeature(new LSPTypeDefinitionFeature());
+        }
+        return typeDefinitionFeature;
+    }
+
+    /**
+     * Initialize the LSP typeDefinition feature.
+     *
+     * @param typeDefinitionFeature the LSP typeDefinition feature.
+     * @return the LSP client features.
+     */
+    public final LSPClientFeatures setTypeDefinitionFeature(@NotNull LSPTypeDefinitionFeature typeDefinitionFeature) {
+        typeDefinitionFeature.setClientFeatures(this);
+        this.typeDefinitionFeature = typeDefinitionFeature;
+        return this;
+    }
+
+    /**
+     * Returns the LSP usage feature.
+     *
+     * @return the LSP usage feature.
+     */
+    @NotNull
+    public final LSPUsageFeature getUsageFeature() {
+        if (usageFeature == null) {
+            setUsageFeature(new LSPUsageFeature());
+        }
+        return usageFeature;
+    }
+
+    /**
+     * Initialize the LSP usage feature.
+     *
+     * @param usageFeature the LSP usage feature.
+     * @return the LSP client features.
+     */
+    public final LSPClientFeatures setUsageFeature(@NotNull LSPUsageFeature usageFeature) {
+        usageFeature.setClientFeatures(this);
+        this.usageFeature = usageFeature;
+        return this;
+    }
+    
+    public void setServerWrapper(LanguageServerWrapper serverWrapper) {

```

Comment:
I think this should be tagged `@ApiStatus.Internal`

Reply:
+1 fixed